events/events.go

```diff
@@ -0,0 +1,205 @@
+package events
+
+import (
+	"time"
+
+	pb "github.com/spacemeshos/api/release/go/spacemesh/v1"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/log"
+)
+
+type UserEvent struct {
+	Event *pb.Event
+}
+
+func EmitBeacon(epoch types.EpochID, beacon types.Beacon) {
+	const help = "Node computed randomness beacon, it will be used to determine eligibility to participate in the consensus."
+	emitUserEvent(
+		help,
+		false,
+		&pb.Event_Beacon{
+			Beacon: &pb.EventBeacon{
+				Epoch:  epoch.Uint32(),
+				Beacon: beacon[:],
+			},
+		},
+	)
+}
+
+func EmitInitStart(smesher types.NodeID, commitment types.ATXID) {
+	const help = "Node started post data initialization. Note that init is noop if node restarted when init was ready."
+	emitUserEvent(
+		help,
+		false,
+		&pb.Event_InitStart{
+			InitStart: &pb.EventInitStart{
+				Smesher:    smesher[:],
+				Commitment: commitment[:],
+			},
+		},
+	)
+}
+
+func EmitInitComplete(failure bool) {
+	const help = "Node completed post data initialization."
+	emitUserEvent(
+		help,
+		failure,
+		&pb.Event_InitComplete{
+			InitComplete: &pb.EventInitComplete{},
+		},
+	)
+}
+
+func EmitPoetWait(current, publish types.EpochID, wait time.Duration) {
+	const help = "Node needs to wait for poet registration window in current epoch to open. " +
+		"Once opened it will submit challenge and wait till poet round ends in publish epoch."
+	emitUserEvent(
+		help,
+		false,
+		&pb.Event_PoetWait{PoetWait: &pb.EventPoetWait{
+			Current: current.Uint32(),
+			Publish: publish.Uint32(),
+			Wait:    durationpb.New(wait),
```

Comment:
I think putting the specific time here (and for other events with the Wait fields) would be better for users. It will, I think be clearer and easier to remember. Also if the UI were to try to convert this duration into the actual time to wait till, it will be off, due to the delay in delivering the event. But the other way around you can always accurately generate the time left from an exact timestamp.

Reply:
there is a timestamp for every event, i think what you suggest can be accomplished using that timestamp on ui if needed

Reply:
I still think an exact time here would be preferable since we do calculate exact times for events (I.E. a layer has an exact time that should be identical for all users) using the timestamp you won't actually be able to calculate an exact time, the time will be off by some amount. Probably a minuscule amount but even so I think the logs would be clearer and, debugging issues with users could be significantly simpler if we could rely on them logging exact times for events, because you could filter or grep by exact time.

Reply:
not all of this events are determined by layer number. both poet events are time based, e.g we have a deadline based on some other logic. for atx this is a wait time till the start of the target epoch, after that user will receive different event

as you said time won't be accurate, but the inaccuracy will be so small that it doesn't matter for human to make any decisions. tbh i am not even sure that smapp should print this wait time, i just added more stuff for @brusherru to consider what to use

so for me the diff between timestamp and duration is so irrelevant that i can't justify going back and change things

Reply:
My point was more regarding making debugging simpler, by having consistent logs for all users. I don't think it will make a difference when it comes to taking decisions. I do however think that having the time to wait is very important for users, since these times could be huge ( on the order of weeks ) we need to manage users expectations.